packages/ketcher-core/src/application/editor/actions/closelyFusing.ts

```diff
@@ -98,3 +98,35 @@ function closestToMerge(struct, closestMap) {
 
   return mergeMap
 }
+
+function omitAtomsAndBondsOfSaltsAndSolventsFromClosestMap(
+  closestMap: MergeMap,
+  sgroups: Map<number, ReSGroup>
+) {
+  const sgroupsOnCanvas = Array.from(sgroups.values()).map(({ item }) => item)
+
+  const mergeMap: MergeMap = {
+    atoms: new Map<number, number>(),
+    bonds: new Map<number, number>()
+  }
+
+  closestMap.atoms.forEach((destinationAtomId, sourceAtomId) => {
+    if (
+      !SGroup.isAtomInSaltOrSolvent(destinationAtomId, sgroupsOnCanvas) &&
+      !SGroup.isAtomInSaltOrSolvent(sourceAtomId, sgroupsOnCanvas)
+    ) {
+      mergeMap.atoms.set(sourceAtomId, destinationAtomId)
+    }
+  })
+
+  closestMap.bonds.forEach((destinationBondId, sourceBondId) => {
+    if (
+      !SGroup.isBondInSaltOrSolvent(destinationBondId, sgroupsOnCanvas) &&
+      !SGroup.isBondInSaltOrSolvent(sourceBondId, sgroupsOnCanvas)
+    ) {
+      mergeMap.bonds.set(sourceBondId, destinationBondId)
+    }
+  })
+
+  return mergeMap
```

Comment:
Maybe we shouldn't separate testing for atoms and bonds involved in merge and cancel merge if there is any sign of salt or solvent at all.
But I'm not sure about all possible consequences, so it's just an assumption for now.